integration_tests/src/main/python/map_test.py

```diff
@@ -32,6 +38,39 @@ def test_simple_get_map_value(data_gen):
                 'a["NOT_FOUND"]',
                 'a["key_5"]'))
 
+@pytest.mark.parametrize('key_gen', [StringGen(nullable=False), IntegerGen(nullable=False), basic_struct_gen], ids=idfn)
+@pytest.mark.parametrize('value_gen', [StringGen(nullable=False), IntegerGen(nullable=False), basic_struct_gen], ids=idfn)
+def test_single_entry_map(key_gen, value_gen):
+    data_gen = [('a', key_gen), ('b', value_gen)]
+    assert_gpu_and_cpu_are_equal_collect(
+            lambda spark : gen_df(spark, data_gen).selectExpr(
+                'map("literal_key", b) as map1',
+                'map(a, b) as map2'))
+
+def test_map_expr_no_pairs():
+    data_gen = [('a', StringGen(nullable=False)), ('b', StringGen(nullable=False))]
+    assert_gpu_and_cpu_are_equal_collect(
+            lambda spark : gen_df(spark, data_gen).selectExpr(
+                'map() as m1'))
+
+def test_map_expr_multiple_pairs():
+    data_gen = [('a', StringGen(nullable=False)), ('b', StringGen(nullable=False))]
```

Comment:
We should add some comments here that because of the high number of values possible in StringGen that in this test duplicate keys never show up.

Reply:
So is there a way/parameter to force generating duplicates?

Reply:
We would have to force StringGen for both a and b to be much more restrictive.

```
 data_gen = [('a', StringGen('duplicate', nullable=False)), ('b', StringGen('duplicate', nullable=False))]
```

Now the only value that will ever be in the 'a' or 'b' column is 'duplicate'. Not that we want to do that, because the test would fail with the CPU run throwing an exception that there are duplicate keys.